### PR TITLE
🌱 Automate go dependency bumps (except CR/k8s.io)

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by sbueringer** [comment]
@vincepri @fabriziopandini I would like to give this a try. I think it should be a good trade-off to bump everything except cr/k8s.io automatically vs. trying to keep up manually.

**Comment by sbueringer** [comment]
cc @killianmuldoon @oscr

File: .github/dependabot.yml [hunk 1]
```diff
@@ -8,13 +11,20 @@ updates:
...
   labels:
     - "ok-to-test"
+# Go
 - package-ecosystem: "gomod"
   directory: "/"
   schedule:
     interval: "weekly"
-  allow:
-    - dependency-name: "github.com/coredns/corefile-migration"
+  ignore:
```
Comment (oscr) [inline .github/dependabot.yml:19]: We should also ignore dependencies that need to be bumped also in the Makefile: like Ginkgo, envsubst etc.

**Comment by oscr** [comment]
Excellent initiative to reduce toil @sbueringer. Overall I think this is great.

File: .github/dependabot.yml (hunk 1 above)
Comment (sbueringer) [reply .github/dependabot.yml:19]: Should be fine if we merge https://github.com/kubernetes-sigs/cluster-api/pull/7503/files before, correct?

**Review by killianmuldoon — APPROVED** [review]
/lgtm

File: .github/dependabot.yml (hunk 1 above)
Comment (oscr) [reply .github/dependabot.yml:19]: Yes, I both hope and think so :)

**Comment by sbueringer** [comment]
/hold
Let's merge #7503 first

**Comment by fabriziopandini** [comment]
+1 to give this a try now that we are just beginning a new release cycle
/lgtm
/approve

**Pull request merged**